packages/slate-editor/src/extensions/blink-nodes/components/BlinkNodes.tsx

```diff
@@ -0,0 +1,67 @@
+import type { RefObject } from 'react';
+import React, { useEffect, useState } from 'react';
+import type { Editor, Node } from 'slate';
+import { ReactEditor, useSlateStatic } from 'slate-react';
+
+import styles from './BlinkNodes.module.scss';
+
+export function BlinkNodes({ containerRef }: { containerRef: RefObject<HTMLDivElement> }) {
+    const editor = useSlateStatic();
+
+    return (
+        <>
+            {editor.nodesToBlink.map(([top, bottom], index) => (
+                <Blinker
+                    key={index}
+                    editor={editor}
+                    top={top}
+                    containerRef={containerRef}
+                    bottom={bottom}
+                    clear={() =>
+                        (editor.nodesToBlink = editor.nodesToBlink.filter((_, i) => i !== index))
+                    }
+                />
+            ))}
+        </>
+    );
+}
+
+function Blinker({
+    editor,
+    top,
+    bottom,
+    containerRef,
+    clear,
+}: {
+    editor: Editor;
+    top: Node;
+    bottom: Node;
+    containerRef: RefObject<HTMLDivElement>;
+    clear: () => void;
```

Comment:
I suggest renaming this prop to have `on` prefix. To match the common callback prop convention. 

Also, `onComplete` seems to be more clear, IMO.